src/sql/src/kafka_util.rs

```diff
@@ -283,7 +301,7 @@ pub fn generate_ccsr_client_config(
         ),
     }
 
-    let mut ccsr_options = extract(
+    let mut ccsr_options = get_and_validate(
```

Comment:
I think we could use `remove_and_validate` here if I'm reading the code right - because afaict we bind `ccsr_options`  to whatever we return from calling get and validate so I don't think we care whether we remove it from the original hashmap at all? and then we can remove `get_and_validate` because this seems like the only callsite?

Reply:
more generally, having both `get_and_validate` and `remove_and_validate` seemed confusing to me and I think we can make it so that we only need one of those two

Reply:
The problem I kept hitting was that the ccsr_options fn param is an immutable borrow, and making it mutable had an annoying trickle up to callers (most of those callers are pretty long functions and I didn't want to expose more mutable state inside the fn body than necessary). Looking at this again, I can take ownership of ccsr_options and it'll keep things nice and simple, so I'll do that and tidy this up. Thanks! 🙂